tests/v2/validation/workloads/workload.go

```diff
@@ -0,0 +1,133 @@
+	log.Info("Waiting deployment comes up active")
+	err := charts.WatchAndWaitDeployments(client, clusterName, namespaceName, metav1.ListOptions{
+		FieldSelector: "metadata.name=" + scaleDeployment.Name,
+	})
+	require.NoError(t, err)
+
+	log.Info("Waiting for all pods to be running")
+	err = pods.WatchAndWaitPodContainerRunning(client, clusterName, namespaceName, scaleDeployment)
+	require.NoError(t, err)
+
+	log.Infof("Counting all pods running by image %s", image)
+	countPods, err := pods.CountPodContainerRunningByImage(client, clusterName, namespaceName, image)
+	require.NoError(t, err)
+	require.Equal(t, expectedReplicas, countPods)
+}
+
+func rollbackDeployment(client *rancher.Client, clusterID, namespaceName string, deploymentName string, revision int) (string, error) {
+	deploymentCmd := fmt.Sprintf("deployment.apps/%s", deploymentName)
+	revisionCmd := fmt.Sprintf("--to-revision=%s", strconv.Itoa(revision))
+	execCmd := []string{"kubectl", "rollout", "undo", "-n", namespaceName, deploymentCmd, revisionCmd}
+	logCmd, err := kubectl.Command(client, nil, clusterID, execCmd, "")
+	return logCmd, err
+}
+
+func verifyDeploymentAgainstRolloutHistory(client *rancher.Client, clusterID, namespaceName string, deploymentName string, expectedRevision string) error {
+	dynamicClient, err := client.GetDownStreamClusterClient(clusterID)
+	if err != nil {
+		return err
+	}
+
+	deploymentResource := dynamicClient.Resource(deployments.DeploymentGroupVersionResource).Namespace(namespaceName)
+
+	unstructuredResp, err := deploymentResource.Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	latestDeployment := &appv1.Deployment{}
+	err = scheme.Scheme.Convert(unstructuredResp, latestDeployment, unstructuredResp.GroupVersionKind())
+	if err != nil {
+		return err
+	}
+
+	if latestDeployment.ObjectMeta.Annotations == nil {
+		return errors.New("revision empty")
+	}
+
+	revision := latestDeployment.ObjectMeta.Annotations[revisionAnnotation]
+
+	if revision == expectedRevision {
+		return nil
+	}
+
+	return errors.New("revision not found")
+}
+
+func verifyOrchestrationStatus(client *rancher.Client, clusterID, namespaceName string, deploymentName string, isPaused bool) error {
+	dynamicClient, err := client.GetDownStreamClusterClient(clusterID)
+	if err != nil {
+		return err
+	}
+
+	deploymentResource := dynamicClient.Resource(deployments.DeploymentGroupVersionResource).Namespace(namespaceName)
+
+	unstructuredResp, err := deploymentResource.Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	latestDeployment := &appv1.Deployment{}
+	err = scheme.Scheme.Convert(unstructuredResp, latestDeployment, unstructuredResp.GroupVersionKind())
+	if err != nil {
+		return err
+	}
+
+	if latestDeployment.Spec.Paused == isPaused {
+		return nil
+	}
+
+	if isPaused {
+		return errors.New("the orchestration is paused")
+	}
+
+	return errors.New("the orchestration is active")
+
```

Comment:
some differences here vs. in the other PRs.

Reply:
There are some differences because in the release/v2.7 branch, I can't use Wrangler, so the code is slightly different.